wordpress-mobile/WordPress-iOS
# Comments: refetch when selected filter changes

Ref: #15955 
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/357

This uses the WPKit changes to specify what Comment `status` to use when fetching Comments. When a filter is selected, Comments are refetched using the `CommentStatusFilter` associated with each tab.

Known issues:
- The UI hasn't been updated yes, so `Trashed` and `Spam` are not displayed correctly.
- Comments are not always refreshed when the view is displayed, so pull to refresh is sometimes needed. 
- `All` is still using the `all` endpoint status (instead of combining Pending and Approved).
- The selected filter is not retained.

To test:
Prerequisite: Enable `commentFilters` feature flag.

- Go to Site > Comments.
- Select all the tabs on the filter tab bar.
- Since Trashed and Spam aren't displayed correctly, I added temp logging in WPKit to see the parameters. The `status` param should correspond with the filter. Examples:

WP sites: 
```
🔴 REST > params: {
    context = edit;
    force = wpcom;
    number = 100;
    status = approved;
```

Self hosted sites:
```
🔴 XMLRPC > params: {
    number = 100;
    status = approve;
}
```



PR submission checklist:
